### PR TITLE
Add SuperAlert

### PR DESCRIPTION
**Pull request by 0832k12**
An extension can open an Alert at the center of screen.
image

### CODE REVIEW

**Comment by Martinelplayz** [comment]
can you make options for the user to press? like (yes, no) (accept, cancel) etc

**Comment by horizontal-shading** [comment]
Why are the blocks in chinese?

**Comment by CST1229** [comment]
> Why are the blocks in chinese?

Probably because the creator is Chinese.

**Comment by SharkPool-SP** [comment]
Can you add more blocks that can customize the alert box? Here is some concept blocks I made (that do work, so this is possible): 
Screen Shot 2023-07-20 at 11 11 52 AM
Heres the .js file with the blocks. If you use this, you don't have to give me credit: 
[New Alert Blocks.js.zip](https://github.com/TurboWarp/extensions/files/12113006/New.Alert.Blocks.js.zip)

**Comment by 0832k12** [comment]
That 's a good advice.

**Comment by 0832k12** [comment]
image
image

**Comment by SharkPool-SP** [comment]
> image image

SICK!!!! How did you get the select color block argument?

**Comment by SurvExe1Pc** [comment]
> > image image
> 
> SICK!!!! How did you get the select color block argument?

`Scratch.ArgumentType.COLOR`

**Comment by SharkPool-SP** [comment]
> > > image image
> > 
> > 
> > SICK!!!! How did you get the select color block argument?
> 
> `Scratch.ArgumentType.COLOR`

Oh ok, thanks!

File: extensions/0832/SuperAlert.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+                            A: {
+                                type: Scratch.ArgumentType.STRING,
+                                menu: 'alignmentOptions',
+                                defaultValue: 'center',
+                            },
+                        },
+                    },
+                ],
+                menus: {
+                    textOptions: [
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:284]: Any reason this isn't acceptreporters: true?

File: extensions/0832/SuperAlert.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+                        { text: 'Title', value: 'Title' },
+                        { text: 'Content', value: 'Content' },
+                    ],
+                    alignmentOptions: [
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:288]: Any reason this isn't acceptreporters: true?

File: extensions/0832/SuperAlert.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                `;
+                    c.style.opacity = "1";
+                    c.style.transform = `translate(${this.xPos}%, ${this.yPos}%) scale(1)`;
+                }, 300);
+            }
+        }
+
+        getInfo() {
+            return {
+                id: 'tc',
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:155]: Extension ID should generally be more descriptive, containing the username of the person who made it and the extension's name. "k12superalert" would be a better ID, for example.

File: extensions/0832/SuperAlert.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                name: Scratch.translate({ id: 'name', default: 'Alert' }),
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:156]: Probably a translation error, but you may want to rename the extension default to 'SuperAlert' to align with the name in the Pull Request.

File: extensions/0832/SuperAlert.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+            this.yPos = -50; // Default Y position
+            this.titleFont = "Arial";
+            this.contentFont = "Arial";
+            this.titleColor = "white";
+            this.contentColor = "white";
+            this.titleSize = 24;
+            this.contentSize = 16;
+            this.outlineSize = 1;
+        }
+        d() {
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:71]: It appears there's no way to open a text box with it's contents already defined; it'll always be empty for a few frames before content is added. Is there a way around this?

File: extensions/0832/SuperAlert.js [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+                    textOptions: [
+                        { text: 'Title', value: 'Title' },
+                        { text: 'Content', value: 'Content' },
+                    ],
+                    alignmentOptions: [
+                        { text: 'Left', value: 'left' },
+                        { text: 'Center', value: 'center' },
+                        { text: 'Right', value: 'right' },
+                    ],
+                    fontOptions: [
```
Comment (NexusKitten) [inline extensions/0832/SuperAlert.js:293]: Be careful with font options, as not all built-in fonts are supported on every device. Preferably, the extension could run off of fonts from the VM, but I'm not sure if that's possible, so the current solution is alright.

**Review by NexusKitten — CHANGES_REQUESTED** [review]
Also change the JS and extensions.json to comply with the new policies in [CONTRIBUTING.md](https://github.com/TurboWarp/extensions/blob/master/CONTRIBUTING.md).
Looking forward to this extension.

**Comment by SharkPool-SP** [comment]
> can you make options for the user to press? like (yes, no) (accept, cancel) etc

I did this in Better Input

**Comment by LilyMakesThings** [comment]
> > can you make options for the user to press? like (yes, no) (accept, cancel) etc
> 
> I did this in Better Input

sir this is the pr for superalert
image

**Comment by SharkPool-SP** [comment]
> > > can you make options for the user to press? like (yes, no) (accept, cancel) etc
> > 
> > 
> > I did this in Better Input
> 
> sir this is the pr for superalert image

ma'am, we are both using CSS. I have no argument but still. Work is already done lmao

**Comment by Cbr2011** [comment]
i like to see this extension be merged.
i also like if every time the user clicks out of the popup or ESC is pressed, The popup closes

**Comment by SharkPool-SP** [comment]
> i like to see this extension be merged.
> i also like if every time the user clicks out of the popup or ESC is pressed, The popup closes

Just use an if statement and close the box if true

**Comment by Cbr2011** [comment]
> > i like to see this extension be merged.
> > i also like if every time the user clicks out of the popup or ESC is pressed, The popup closes
> 
> Just use an if statement and close the box if true

oh ok

**Comment by SurvExe1Pc** [comment]
guys we love XSS

**Comment by SurvExe1Pc** [comment]
i mean this pull seems dead but this extension has XSS in it.
image

**Comment by CubesterYT** [comment]
W xss /j

**Comment by CubesterYT** [comment]
Why was this closed?

**Pull request closed without merging**